### PR TITLE
B21575-main3-add_order_types

### PR DESCRIPTION
**Pull request by JamesHawks224**
## Agility ticket
[B21575](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21575)

## INT PR
[14211](https://github.com/transcom/mymove/pull/14211)

## Summary/Testing/ANDI/Screenshots
* See INT PR's

### CODE REVIEW

**Review by WeatherfordAaron — APPROVED** [review]
found no diff between INT and MAIN PRs. lgtm 👽

**Review by taeJungCaci — CHANGES_REQUESTED** [review]
looks like there are some linter fixes needed in couple of these files. Other than that - LGTM

**Comment by WeatherfordAaron** [comment]
> looks like there are some linter fixes needed in couple of these files. Other than that - LGTM

Agreed. I think the Happo is showing something broken for the File Viewer or something too.. at least 1 part of this one may get resolved by updating the branch.

**Comment by JamesHawks224** [comment]
> looks like there are some linter fixes needed in couple of these files. Other than that - LGTM

I believe I've resolved those.  Please re-evaluate.

**Review by taeJungCaci — APPROVED** [review]

**Comment by WeatherfordAaron** [comment]
Note: There is a current Happo fail on this PR. The fix is in another PR
https://github.com/transcom/mymove/pull/14303

This is from Agility Ticket [B-21373](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21373) which has been merged into INT but is sitting in the Testing lane.

**Comment by antgmann** [comment]
Please link all integration PRs that this main PR is covering. The only int pr linked covers just... wait. the commits between the linked PR and this PR don't match up?
[a6077ee](https://github.com/transcom/mymove/pull/14206/commits/a6077ee652d3c5b7ac598fda4e3fd0b5a73ec123) matches.
[ce85e0f](https://github.com/transcom/mymove/pull/14206/commits/ce85e0fcec14121d8e9f96e825578fcbf98013b0) and [b3d6905](https://github.com/transcom/mymove/pull/14211/commits/b3d69055d828f1aa6c68bc8cae19f679af64d008) don't, and none of them seem to match exactly again until commit [57f970d](https://github.com/transcom/mymove/pull/14206/commits/57f970d9f89b716bc15a023dfa0fd748b7478110). Am I misunderstanding something?

File: src/components/Customer/EditOrdersForm/EditOrdersForm.jsx [hunk 1]
```diff
@@ -165,7 +173,7 @@ const EditOrdersForm = ({
         new_duty_location: true,
       }}
     >
-      {({ isValid, isSubmitting, handleSubmit, setValues, setFieldValue, values }) => {
+      {({ isValid, isSubmitting, handleSubmit, handleChange, setValues, values, setFieldValue }) => {
         const isRetirementOrSeparation = ['RETIREMENT', 'SEPARATION'].includes(values.orders_type);
```
Comment (antgmann) [inline src/components/Customer/EditOrdersForm/EditOrdersForm.jsx:177]: Discrepancy 
image

File: src/components/Customer/EditOrdersForm/EditOrdersForm.jsx [hunk 2]
```diff
@@ -184,18 +192,39 @@ const EditOrdersForm = ({
...
-            setShowAccompaniedTourField(true);
-            setShowDependentAgeFields(true);
+          if (e.target.value === '') {
+            setFieldValue('has_dependents', '');
           } else {
-            setShowAccompaniedTourField(false);
-            setShowDependentAgeFields(false);
+            const fieldValueHasDependents = e.target.value === 'yes';
+            setHasDependents(fieldValueHasDependents);
+            setFieldValue('has_dependents', fieldValueHasDependents ? 'yes' : 'no');
```
Comment (antgmann) [inline src/components/Customer/EditOrdersForm/EditOrdersForm.jsx:200]: Discrepancy
image

File: src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx [hunk 3]
```diff
@@ -36,6 +36,11 @@ const OrdersInfoForm = ({ ordersTypeOptions, initialValues, onSubmit, onBack })
   const [hasDependents, setHasDependents] = useState(false);
   const [isOconusMove, setIsOconusMove] = useState(false);
   const [enableUB, setEnableUB] = useState(false);
+
+  const [isHasDependentsDisabled, setHasDependentsDisabled] = useState(false);
+  const [prevOrderType, setPrevOrderType] = useState('');
+  const [filteredOrderTypeOptions, setFilteredOrderTypeOptions] = useState(ordersTypeOptions);
+
```
Comment (antgmann) [inline src/components/Customer/OrdersInfoForm/OrdersInfoForm.jsx:43]: Small spacing discrepancy

File: src/components/Office/AddOrdersForm/AddOrdersForm.jsx [hunk 4]
```diff
@@ -13,7 +13,7 @@ import ToolTip from 'shared/ToolTip/ToolTip';
 import { DatePickerInput, DropdownInput, DutyLocationInput } from 'components/form/fields';
 import { Form } from 'components/form/Form';
 import SectionWrapper from 'components/Customer/SectionWrapper';
-import { ORDERS_PAY_GRADE_OPTIONS } from 'constants/orders';
+import { ORDERS_PAY_GRADE_OPTIONS, ORDERS_TYPE } from 'constants/orders';
 import { dropdownInputOptions } from 'utils/formatters';
```
Comment (antgmann) [inline src/components/Office/AddOrdersForm/AddOrdersForm.jsx:17]: This file is not edited on int pr https://github.com/transcom/mymove/pull/14211/files or https://github.com/transcom/mymove/pull/14378/files, or even https://github.com/transcom/mymove/pull/14164/files

File: migrations/app/schema/20241111221400_add_new_order_types.up.sql [hunk 5]
```diff
@@ -0,0 +1,5 @@
+-- adding order types to orders_type enum used in the orders table.
```
Comment (antgmann) [inline migrations/app/schema/20241111221400_add_new_order_types.up.sql:1]: image

Nothing to do with this migration, but this change to scripts/db-truncate also isn't present. Probably just the linter giving you a hard time?

**Review by antgmann — CHANGES_REQUESTED** [review]
~~Okay I'm definitely confused on what's going on with this PR, there's some substantial differences between this one and the INT ones. Commits don't line up, and changes aren't the exact same either. 

I hate to throw a wrench into this, but it's important things match between environments. See review comments for more info and let me know if there's anything I can do to help get things back on track.~~

After some discussion, the only remaining issue is the following commit: [934f477](https://github.com/transcom/mymove/pull/14206/commits/934f4772ed0b7f4fb3309f5b99913a4605ead644)

**Comment by taeJungCaci** [comment]
> Okay I'm definitely confused on what's going on with this PR, there's some substantial differences between this one and the INT ones. Commits don't line up, and changes aren't the exact same either.
> 
> I hate to throw a wrench into this, but it's important things match between environments. See review comments for more info and let me know if there's anything I can do to help get things back on track.

Trying to see if I can help, but I've created a new branch off the IntergrationBranch just now and merged in this branch into it to see. [This is the PR](https://github.com/transcom/mymove/pull/14395/files). I only see 1 diff which is the `src/components/Office/AddOrdersForm/AddOrdersForm.jsx` that was actually added after I approved this PR. Not sure why on that though and glad you caught that.

**Comment by antgmann** [comment]
> Trying to see if I can help, but I've created a new branch off the IntergrationBranch just now and merged in this branch into it to see. [This is the PR](https://github.com/transcom/mymove/pull/14395/files). I only see 1 diff which is the `src/components/Office/AddOrdersForm/AddOrdersForm.jsx` that was actually added after I approved this PR. Not sure why on that though and glad you caught that.

This is very helpful as it lets us see that all of the discrepancies I listed above (aside from # 4, that's the additional commit added that you mention) are actually in integrationTesting, even if it's confusing how exactly that happened. Seems like maybe we've lost an int pr somewhere along the way? 

Either way, it implies that this should be a much more solid PR once [934f477](https://github.com/transcom/mymove/pull/14206/commits/934f4772ed0b7f4fb3309f5b99913a4605ead644) is added to integration first.

**Comment by JamesHawks224** [comment]
> > Trying to see if I can help, but I've created a new branch off the IntergrationBranch just now and merged in this branch into it to see. [This is the PR](https://github.com/transcom/mymove/pull/14395/files). I only see 1 diff which is the `src/components/Office/AddOrdersForm/AddOrdersForm.jsx` that was actually added after I approved this PR. Not sure why on that though and glad you caught that.
> 
> This is very helpful as it lets us see that all of the discrepancies I listed above (aside from # 4, that's the additional commit added that you mention) are actually in integrationTesting, even if it's confusing how exactly that happened. Seems like maybe we've lost an int pr somewhere along the way?
> 
> Either way, it implies that this should be a much more solid PR once [934f477](https://github.com/transcom/mymove/pull/14206/commits/934f4772ed0b7f4fb3309f5b99913a4605ead644) is added to integration first.

I think this will help clarify why things seem so odd.  
* my main branch: completed dev
* created my INT branch and merged my main branch into it
* B21374 was in INT.  It introduced changes in the same code of my changes.  Because B21374 was not in main yet, I could not make changes in my main and then merge to my int.  
* I made changes in the INT branch to get it to work with the B21374 changes. 
* When B21374 made it to main, I then updated my main (which was essentially the same changes in my INT).  
* Because I made changes to my main, I created another INT pr to apply that commit to INT per Daniel's instructions.

**Comment by taeJungCaci** [comment]
> > Trying to see if I can help, but I've created a new branch off the IntergrationBranch just now and merged in this branch into it to see. [This is the PR](https://github.com/transcom/mymove/pull/14395/files). I only see 1 diff which is the `src/components/Office/AddOrdersForm/AddOrdersForm.jsx` that was actually added after I approved this PR. Not sure why on that though and glad you caught that.
> 
> This is very helpful as it lets us see that all of the discrepancies I listed above (aside from # 4, that's the additional commit added that you mention) are actually in integrationTesting, even if it's confusing how exactly that happened. Seems like maybe we've lost an int pr somewhere along the way?
> 
> Either way, it implies that this should be a much more solid PR once [934f477](https://github.com/transcom/mymove/pull/14206/commits/934f4772ed0b7f4fb3309f5b99913a4605ead644) is added to integration first.

Yes, or deleted from this PR because it doesn't look like it belong in this BL.

**Comment by antgmann** [comment]
> I think this will help clarify why things seem so odd.
> 
> * my main branch: completed dev
> * created my INT branch and merged my main branch into it
> * B21374 was in INT.  It introduced changes in the same area as my changes.  Because B21374 was not in main yet, I could not make changes in my main and then merge to my int.
> * I made changes in the INT branch to get it to work with the B21374 changes.
> * When B21374 made it to main, I then updated my main with essentially the same changes in my INT.
> * Because I made changes to my main, I created another INT pr to apply that commit to INT per Daniel's instructions.

That'll do it. I'm sorry that you've had to deal with so many conflicts, especially after dev was completed. That makes organizing by dependency much more difficult and harder to follow.

In that case, never mind any of the other differences. It'd be an exercise in futility to track them all down if they're the result of conflicts like that. All of the changes match except for [934f477](https://github.com/transcom/mymove/pull/14206/commits/934f4772ed0b7f4fb3309f5b99913a4605ead644), then this will be g2g without that commit like @taeJungCaci suggests, or once the commit is pushed to main.

**Review by taeJungCaci — CHANGES_REQUESTED** [review]
Requesting change as mentioned above

**Comment by JamesHawks224** [comment]
> Requesting change as mentioned above



> > I think this will help clarify why things seem so odd.
> > 
> > * my main branch: completed dev
> > * created my INT branch and merged my main branch into it
> > * B21374 was in INT.  It introduced changes in the same area as my changes.  Because B21374 was not in main yet, I could not make changes in my main and then merge to my int.
> > * I made changes in the INT branch to get it to work with the B21374 changes.
> > * When B21374 made it to main, I then updated my main with essentially the same changes in my INT.
> > * Because I made changes to my main, I created another INT pr to apply that commit to INT per Daniel's instructions.
> 
> That'll do it. I'm sorry that you've had to deal with so many conflicts, especially after dev was completed. That makes organizing by dependency much more difficult and harder to follow.
> 
> In that case, never mind any of the other differences. It'd be an exercise in futility to track them all down if they're the result of conflicts like that. All of the changes match except for [934f477](https://github.com/transcom/mymove/pull/14206/commits/934f4772ed0b7f4fb3309f5b99913a4605ead644), then this will be g2g without that commit like @taeJungCaci suggests, or once the commit is pushed to main.

Removed the "934f477" commit.

**Review by taeJungCaci — APPROVED** [review]

**Review by antgmann — APPROVED** [review]

**Pull request merged**